cmd/create/cluster/cmd.go

```diff
@@ -1552,16 +1610,39 @@ func run(cmd *cobra.Command, _ []string) {
 	privateSubnetsCount := 0
 
 	var availabilityZones []string
+	var subnets []*ec2.Subnet
+	exclude := false
 	if useExistingVPC || subnetsProvided {
-		subnets, err := awsClient.GetSubnetIDs()
+		initialSubnets, err := awsClient.GetSubnetIDs()
 		if err != nil {
 			r.Reporter.Errorf("Failed to get the list of subnets: %s", err)
 			os.Exit(1)
 		}
 		if subnetsProvided {
 			useExistingVPC = true
 		}
+		_, machineNetwork, err := net.ParseCIDR(machineCIDR.String())
+		if err != nil {
+			r.Reporter.Errorf("Unable to parse machine CIDR")
+			os.Exit(1)
+		}
+		for _, subnet := range initialSubnets {
+			subnetIP, _, err := net.ParseCIDR(*subnet.CidrBlock)
+			if err != nil {
+				r.Reporter.Errorf("Unable to parse subnet CIDR")
+				os.Exit(1)
+			}
+
+			if machineNetwork.Contains(subnetIP) {
+				subnets = append(subnets, subnet)
+			} else {
+				exclude = true
+			}
+		}
 
+		if exclude {
```

Comment:
This could also be len(subnets) != len(initialSubnets), so less branch prediction is required :thinking: